nyxtools/flyer.py

```diff
@@ -20,93 +23,183 @@ def __init__(self, vector, zebra, detector=None) -> None:
         self.zebra = zebra
         self.detector = detector
 
+        self.data_directory_name = None
+        self.file_prefix = None
+        self.num_images = None
+        self.file_number_start = None
+
         self._asset_docs_cache = deque()
-        self._resource_uids = []
-        self._datum_counter = None
-        self._datum_ids = DEFAULT_DATUM_DICT
+        self._resource_document = None
+        self._datum_factory = None
+
+        # self._resource_uids = []
+        # self._datum_counter = None
+        # TODO: rework to use datum ids dictionary.
+        # self._datum_ids = DEFAULT_DATUM_DICT
+        self._datum_ids = []
         self._first_file = None
 
         self._collection_dictionary = None
 
+    def update_parameters(self, **kwargs):
+        self.data_directory_name = kwargs.get("data_directory_name", "/nyx-data/test")
+        self.file_prefix = kwargs.get("file_prefix", "test")
+        self.num_images = kwargs.get("num_images", 1)
+        self.file_number_start = kwargs.get("file_number_start", 1)
+
+        super().update_parameters(**kwargs)
+
     def kickoff(self):
         self.detector.stage()
-        self.vector.move()
 
-        return NullStatus()
+        def zebra_callback(*args, **kwargs):
+            print(f"\n{ttime.ctime()}:\n\targs: {args}\n\tkwargs: {kwargs}\n")
+            self.zebra.pc.arm_signal.put(1)
+            return NullStatus()
+
+        st = self.vector.move()
+        st.add_callback(zebra_callback)
+
+        return st
+
+    def complete(self):
+        st_vector = self.vector.track_move()
+
+        def detector_callback(value, old_value, **kwargs):
+            print(f"\n{ttime.ctime()}: DETECTOR status {old_value} -> {value}\n")
+            # if old_value == "Acquiring" and value == "Done":
+            if old_value == 1 and value == 0:
+                print(f"{ttime.ctime()}: DETECTOR status successfully changed {old_value} -> {value}")
+                return True
+            else:
+                print(f"{ttime.ctime()}: DETECTOR status changing {old_value} -> {value}...")
+                return False
+
+        # token = self.detector.cam.acquire.subscribe(detector_callback)
+        # st_detector = self.detector.cam.acquire._status
+
+        st_detector = SubscriptionStatus(self.detector.cam.acquire, detector_callback, run=True)
+
+        return st_vector & st_detector
+
+    def describe_collect(self):
+        return_dict = {
+            "primary": {
+                f"{self.detector.name}_image": {
+                    "source": f"{self.detector.name}_image",
+                    "dtype": "array",
+                    "shape": [-1, -1],  # TODO: read shapes from AD
```

Comment:
I can do this, unless you're already on it @mrakitin

Reply:
Sounds great! Please have a look at https://github.com/bluesky/databroker/pull/596/files#diff-f7833108df15eaaac29030bc84ed7b307eb03b4eb91d959b4917d94ee1d13004R2097-R2101. That link has a few comments about what shape we expect (`(num_images, y, x)`).